deployments/helm-chart/values.yaml

```diff
@@ -310,3 +310,6 @@ prometheus:
 
   ## Configures the port to scrape the metrics.
   port: 9113
+
+  ## Specifies the namespace/name of a Kubernetes TLS Secret which will be used to protect the Prometheus endpoint.
+  secret: ""
```

Comment:
could we add the ability to configure the secret values? this is handy, as it allows the admins to define every part of the Ingress Controller installation through a helm chart. 

> For Helm chart, we can add parameters similar to the parameters for the default server secret and the wildcard secret ( see `controller.wildcardTLS.*`). There is no need to provide the default value of the cert and key. Note that we already have `prometheus.*` parameter, so it makes sense to extend them. For example, prometheus.tls.cert, prometheus.tls.key and prometheus.tls.secret. If the prometheus.tls.secret, helm will configure the IC args to use the referenced secret. If `prometheus.tls.cert` and `prometheus.tls.key` are set, helm will create a Secret with those values and configure the IC args to use it.
From https://github.com/nginxinc/kubernetes-ingress/pull/1652#issuecomment-857284938

Reply:
there are a few ways I see to keep those values protected:
- a plugin to encrypt secret values https://itnext.io/helm-3-secrets-management-4f23041f05c3 
- have a separate values files only for secrets (helm install can take multiple values , for examle, `helm install .. -f a.yaml -f b.yaml -f c.yaml`) and not store it in the source control, but generate it during deployment 
- you can also configure all or some helm parameters via cli args, so helm can take the values for those parameters from env variables.

Allowing configuring secret values in the helm chart allows the admin to use the chart as a single source of truth -- so to install the Ingress Controller, you only need to run `helm install ... ` and nothing else.

I think it is important to provide an alternative, so that secrets could be managed separately. That's why the `secret` parameter makes sense.

Reply:
I understand there are mitigations, such as additional values files and encryption and cli-args. 

I still think it's not a good idea and I won't be adding secret values to this PR. If you'd like to block this PR based on this, let me know so we can do something to resolve it.